src/NServiceBus.Core.Tests/Routing/MessageDrivenUnsubscribeTerminatorTests.cs

```diff
@@ -59,7 +63,10 @@ public void Should_Throw_when_max_retries_reached()
             state.RetryDelay = TimeSpan.Zero;
             dispatcher.FailDispatch(11);
 
-            Assert.That(async () => await terminator.Invoke(new UnsubscribeContext(new FakeContext(), typeof(object), options), c => TaskEx.CompletedTask), Throws.InstanceOf<QueueNotFoundException>());
+            var context = new TestableUnsubscribeContext();
+            context.Extensions = options.Context;
```

Comment:
this seems to be not that uncommon. Would it be interesting to provide some helper to merge SendOptions into the context? e.g. `context.Merge(sendOptions);`